### PR TITLE
feature/OE-205-cascade-office-region-department-deletion-to-daily-numbers

### PR DESCRIPTION
**Pull request by devsquad-mathews-lima**
## Review

For the submitter, initial self-review:

- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency between the changeset and the goal stated above
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Complexity - is the proposed change introducing any unjustified complexity? 
- [ ] Thoroughness of added tests and any missing edge cases

### CODE REVIEW

File: app/Http/Livewire/Castle/Departments.php [hunk 1]
```diff
@@ -59,11 +64,25 @@ public function destroy()
...
+                'user.office',
+                fn(Builder $query) => $query->whereIn('id', $officeIds)
+            )->delete();
+
+            Office::whereIn('id', $officeIds)->delete();
+
+            $department->regions()->delete();
+
+            $department->delete();
+        });
```
Comment (Ron-Junior) [inline app/Http/Livewire/Castle/Departments.php:85]: About the users inside of a department... these users will be softdeleted? or will have a department_id null?

File: app/Http/Livewire/Castle/Offices.php [hunk 2]
```diff
@@ -70,14 +72,20 @@ public function destroy()
...
 
-        $office->delete();
+        DB::transaction(function () use ($office) {
+            DailyNumber::whereHas(
+                'user.office',
+                fn(Builder $query) => $query->whereIn('id', [$office->id])
+            )->delete();
+
+            $office->delete();
+        });
```
Comment (Ron-Junior) [inline app/Http/Livewire/Castle/Offices.php:88]: The same here. Maybe in this case the users will be an office_id null instead off sofdeleted.

File: app/Http/Livewire/Castle/Regions.php [hunk 3]
```diff
@@ -66,14 +68,21 @@ public function destroy()
...
-        $region->delete();
+        DB::transaction(function () use ($region) {
+            DailyNumber::whereHas(
+                'user.office',
+                fn (Builder $query) => $query->whereIn('id', $region->offices->pluck('id'))
+            )->delete();
+
+            $region->offices()->delete();
+            $region->delete();
+        });
```
Comment (Ron-Junior) [inline app/Http/Livewire/Castle/Regions.php:85]: The same here. Maybe in this case the users will be an office_id null instead of sofdeleted.

**Review by Ron-Junior — CHANGES_REQUESTED** [review]

**Review by Ron-Junior — APPROVED** [review]
